frontend/app/src/MetricsManager.ts

```diff
@@ -113,6 +123,45 @@ export class MetricsManager {
     this.appHash = appHash
   }
 
+  // Set metrics url if sent by the host_config, otherwise fallback to retrieving the default
+  public setMetricsConfig = (metricsUrl = ""): void => {
+    // Don't send metrics if explicitly set to "off"
+    if (metricsUrl === "off") {
+      return
+    } else if (metricsUrl) {
+      this.metricsUrl = metricsUrl
+    } else {
+      this.requestDefaultMetricsConfig()
+    }
+  }
+
+  // Fallback - Checks if cached in localStorage, otherwise fetches the config from a default URL
+  private async requestDefaultMetricsConfig(): Promise<any> {
```

Comment:
Oh, I think there might be a small issue here: we probably should not request the `metrics.json` if the app developer has deactivated `gatherUsageStats` but we don't know this info yet when setMetricsConfig is called in `onHostConfigResp` 🤔

Reply:
One way to solve this would be to lazy-load the metrics config in the `initialize` method. Either by only setting the metricsUrl value in `setMetricsConfig` and executing `requestDefaultMetricsConfig` in `initialise` based on that  value. Or we could include move the `metricsUrl` into AppConfig, and pass it in to the `metricsMgr.initialize` method in `handleOneTimeInitialization` via `this.state.appConfig.metricsUrl`

Reply:
Also see a potential third option: check `gatherUsageStats` config in the `host_config` endpoint instead of waiting for it to be passed in NewSession proto, and set `this.actuallySendMetrics` in  `setMetricsConfig`. This would consolidate related logic but a bit more of a refactor. Thoughts?

Reply:
That sounds like a clever solution for this as well 👍 There is one potential concern: What if the platforms defines a static metrics URL in host config